tests/test_other.py

```diff
@@ -8219,6 +8219,30 @@ def test(infile, source_map_added_dir=''):
     ensure_dir('inner')
     test('inner/a.cpp', 'inner')
 
+  def test_emsymbolizer(self):
+    # Test DWARF output
+    self.run_process([EMCC, test_file('core/test_dwarf.c'),
+                      '-g', '-O1', '-o', 'test_dwarf.js'])
+
+    # Use hard-coded addresses. This is potentially brittle, but LLVM's
+    # O1 output is pretty minimal so hopefully it won't break too much?
+    # Another option would be to disassemble the binary to look for certain
+    # instructions or code sequences.
+
+    def get_addr(address):
+      return self.run_process(
+          [PYTHON, path_from_root('emsymbolizer.py'), 'test_dwarf.wasm', address],
+          stdout=PIPE).stdout
+
+    # Check a location in foo(), not inlined.
+    self.assertIn('test_dwarf.c:6:3', get_addr('0x101'))
```

Comment:
We could do a range here perhaps? 0x101 += 10 seems safe and good enough.

Reply:
The problem is that this test is the other way around... i.e. we enter the address and check the line number. Come to think of it, it actually does have some slack in the positive direction, since the line record covers 5 bytes of instructions.

Reply:
The thing that made me worry a little less is that the code looks pretty minimal and not too subject to random changes in e.g. optimizers. Reordering of functions, removal of `__wasm_call_ctors`, changes in imports/exports etc could still break it.

Reply:
Interesting, what's this about 5 bytes? Is the line section limited to that resolution?

What I was suggesting is something like this, can't it work?
```python
for i in range(-10, 10):
  if 'test_dwarf.c:6:3' in get_addr(str(0x101 + i)):
    break
else:
  self.assert('not found')
```

Reply:
There are just 5 bytes of instructions that are considered to be part of line 6 (the call, and the drop of the return value).
I guess that suggestion could work. Although I feel like that would cover small perturbations in the code generation, but probably not changes like the ones I mentioned above (well, I guess it would survive removing or moving __wasm_call_ctors). Another option would be to disassemble the binary and find particular instructions, which would be precise but is potentially almost as error-prone as the code.

Reply:
Disassembling sounds like overkill to me. Sgtm to land this and see if it's an issue in practice.

Reply:
OK, I expect I'll be thinking more about this as I write more tests for the various ways to get line/symbol info.